src/sql/workbench/browser/designer/designer.ts

```diff
@@ -537,6 +537,8 @@ export class Designer extends Disposable implements IThemable {
 		isMainView: boolean): DesignerUIComponent {
 		const propertyPath = [...parentPath, componentDefinition.propertyName];
 		let component: DesignerUIComponent;
+		const hasDescription: boolean = componentDefinition?.description ? true : false;
```

Comment:
we don't need these 2 variables, like I mentioned the other day, we should only subscribe to the onFocus event when isMainView === false.

Reply:
The reason I have this hasDescription variable is because the designer class is generic. It would work now, but what if in future there are other features that use designer and have properties that have no descriptions? And if we select a property with no description after clicking on one with a description, it should update the UI, which is why we need this.

Reply:
I think I have replied to this question (about property without description) in the other comment, my point here is the event should only be subscribed when we are in properties pane.

Reply:
the next line does that

Reply:
![image](https://user-images.githubusercontent.com/13777222/142047977-60b499a9-627d-45e7-b85c-7b19a96a9bb7.png)

you are subscribing to the event without checking whether it is in properties pane

Reply:
check next line
```const showDescription = hasDescription && !isMainView```
in every function call,
```this._propertiesPane.updateDescription(showDescription, componentDefinition)```

Reply:
i mean you should do this, we don't want to subscribe to the event if it is not in properties pane:

```
if(!isMainView){
    dropdown.onDidFocus(() => {
    this._propertiesPane.updateDescription(componentDefinition);
});
}


```